src/alert/alert.spec.ts

```diff
@@ -21,79 +22,179 @@ describe('ngb-alert', () => {
+      expect(getCloseButton(alertEl)).toBeTruthy();
+
+      getCloseButton(alertEl).click();
+      fixture.detectChanges();
+      expect(getAlertElement(fixture.nativeElement)).toBeNull();
+    });
+
+    it('should open a dismissible alert with a specified type', () => {
+      const fixture = createTestComponent(`<template ngbAlert type="success">Hello, {{name}}!</template>`);
+      const alertEl = getAlertElement(fixture.nativeElement);
+
+      expect(alertEl).toHaveCssClass('alert-success');
+      expect(getCloseButton(alertEl)).toBeTruthy();
+    });
+
+    it('should dismiss alert and invoke close handler on close button click', () => {
+      const fixture = createTestComponent(`<template ngbAlert (close)="closed = true">Hello, {{name}}!</template>`);
+      const alertEl = getAlertElement(fixture.nativeElement);
 
-    getCloseButton(alertEl).click();
-    fixture.detectChanges();
+      getCloseButton(alertEl).click();
+      fixture.detectChanges();
 
-    expect(fixture.componentInstance.closed).toBeTruthy();
-    expect(getAlertElement(fixture.nativeElement)).toBeNull();
+      expect(fixture.componentInstance.closed).toBeTruthy();
+      expect(getAlertElement(fixture.nativeElement)).toBeNull();
+    });
+
+    it('should auto close on timeout specified', fakeAsync(() => {
+         const fixture =
+             createTestComponent(`<template ngbAlert [dismissOnTimeout]="1000">Hello, {{name}}!</template>`);
+         const alertEl = getAlertElement(fixture.nativeElement);
+
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(800);
+         fixture.detectChanges();
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(1200);
+         fixture.detectChanges();
+         expect(getAlertElement(fixture.nativeElement)).toBeNull();
+       }));
   });
 
-  it('should auto close on timeout specified', fakeAsync(() => {
-       const fixture = createTestComponent(`<template ngbAlert [dismissOnTimeout]="1000">Hello, {{name}}!</template>`);
-       const alertEl = getAlertElement(fixture.nativeElement);
+  describe('Custom config', () => {
 
-       expect(alertEl.getAttribute('role')).toEqual('alert');
-       expect(getCloseButton(alertEl)).toBeTruthy();
+    beforeEach(() => { TestBed.configureTestingModule({declarations: [TestComponent], imports: [NgbAlertModule]}); });
 
-       tick(800);
-       fixture.detectChanges();
-       expect(alertEl.getAttribute('role')).toEqual('alert');
-       expect(getCloseButton(alertEl)).toBeTruthy();
+    it('should initialize inputs with provided config', () => {
+      let config: NgbDismissibleAlertConfig;
+      const fixture = createTestComponent(`<template ngbAlert>Hello, {{name}}!</template>`);
+      inject([NgbDismissibleAlertConfig], (c: NgbDismissibleAlertConfig) => {
```

Comment:
This seems weird to me - does it need to be injected here? Can you explain a little more in detail about what situation this is simulating?